src/pages/GiftAid/GiftAid.js

```diff
@@ -108,34 +114,34 @@ function GiftAid(props) {
 
   /**
    * Updates validation state for form fields
-   * @param childState
-   * @param name
+   * @param thisFieldsState
+   * @param thisFieldsName
    */
-  const setFieldValidity = (childState, name) => {
-    const prevStateField = fieldValidation[name];
-    const fieldUndefined = prevStateField === undefined;
-    const valueUndefined = typeof prevStateField !== 'undefined' && prevStateField.value === undefined;
-    const newValue = typeof prevStateField !== 'undefined' && prevStateField.value !== childState.value;
-    const newState = (fieldUndefined === false && newValue) || (valueUndefined === true || newValue);
-
+  const setFieldValidity = (thisFieldsState, thisFieldsName) => {
```

Comment:
Renaming a bunch of badly named vars to help me, and hopefully others, understand this a little easier.